app/js/ZipManager.js

```diff
@@ -115,12 +119,32 @@ async function generateZip(projectId, zipfile) {
   zipfile.end()
 }
 
-async function exportProject(projectId) {
-  var zipfile = new yazl.ZipFile()
-  generateZip(projectId, zipfile) // generate zip file in background
-  return zipfile.outputStream
-}
+async function exportProject(projectId, res) {
+  const tmpdir = await fs.promises.mkdtemp(await fs.promises.realpath(os.tmpdir()) + path.sep);
+  const tmpfile = path.join(tmpdir, 'export-project.zip')
+
+  try {
+    const outputStream = fs.createWriteStream(tmpfile)
+
+    const zipfile = new yazl.ZipFile()
 
-module.exports = {
-  exportProject: util.callbackify(exportProject),
+    // start generating the zip contents asynchronously (no await)
+    generateZip(projectId, zipfile).catch(error => {
+      throw error
```

Comment:
This is not going to work. The thrown error will get caught by the `.catch()` handler and become an unhandled promise rejection.

Reply:
The intention is that the `exportProject` promise gets rejected when the error is thrown, so the error is passed to `next` via the route handler.